### PR TITLE
Custom router wan't used

### PR DESCRIPTION
**Pull request by maxmalov**
## Related Issue

https://github.com/inversify/InversifyJS/issues/486

## Motivation and Context

https://github.com/inversify/InversifyJS/issues/486

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/server.ts [hunk 1]
```diff
@@ -96,9 +96,11 @@ export class InversifyExpressServer  {
                     router[metadata.method](metadata.path, ...metadata.middleware, handler);
                 });
 
-                this._app.use(controllerMetadata.path, ...controllerMetadata.middleware, router);
+                this._router.use(controllerMetadata.path, ...controllerMetadata.middleware, router);
             }
         });
+
+        this._app.use("/", this._router);
```
Comment (maxmalov) [inline src/server.ts]: Probably hardcoding `"/"` here isn't the best part. Usually APIs have root namespace such as `/api` or smth. So probably additional configuration is needed. What do you guys think about that?

Comment (lholznagel) [inline src/server.ts]: I totally agree with that.

File: README.md [hunk 2]
```diff
@@ -105,6 +105,16 @@ server.setErrorConfig((app) => {
 });
 ```
 
+### `.setRoutingConfig(routingConfig)`
```
Comment (remojansen) [inline README.md]: Can you please add a note about the default value `{  rootPath: "/"  }` being used here.

File: src/server.ts [hunk 3]
```diff
@@ -25,6 +26,9 @@ export class InversifyExpressServer  {
     ) {
         this._container = container;
         this._router = customRouter || express.Router();
+        this._routingConfig = {
+            rootPath: "/"
```
Comment (remojansen) [inline src/server.ts]: You probably need an extra additional optional argument for the default value.

File: src/server.ts (hunk 1 above)
Comment (remojansen) [inline src/server.ts]: 👍

**Comment by remojansen** [comment]
@maxmalov thanks for the PR I have added a couple of comments after doing the code review

File: src/server.ts (hunk 3 above)
Comment (maxmalov) [inline src/server.ts]: I'm not sure I've got your point here, do you mean provide default routing config or default root path via optional constructor argument?

Comment (remojansen) [inline src/server.ts]: I think it is a good idea to add an optional constructor argument for routingConfig. I know they can use `setRoutingConfig` as well but I feel like is nicer to be able to configure everything via the constructor. What do you think? Thanks!

Comment (maxmalov) [inline src/server.ts]: Yeah, that sounds good to me. Besides there is no need for `setRoutingConfig` in this approach :ok_hand:

File: src/server.ts [hunk 4]
```diff
@@ -96,9 +101,11 @@ export class InversifyExpressServer  {
                     router[metadata.method](metadata.path, ...metadata.middleware, handler);
                 });
 
-                this._app.use(controllerMetadata.path, ...controllerMetadata.middleware, router);
+                this._router.use(controllerMetadata.path, ...controllerMetadata.middleware, router);
```
Comment (maxmalov) [inline src/server.ts]: Oh, missed this newly created `router`. It should be replaced with custom router too.

**Comment by maxmalov** [comment]
@remojansen I've updated PR accroding to requested changes :point_up:

**Comment by remojansen** [comment]
@maxmalov thanks a lot for this PR!

**Pull request merged**